### PR TITLE
Add image index

### LINKED ISSUE #3: Image Index
#### As a user I want the homepage to display all saved images.

__Story__:

Now that we are saving images for our users, we want them to be able to view
the list of images in the system. You might be thinking, "what if my users only
want to view images of cats", or "what if my users only want to see the images
they uploaded". Those are great questions. Maybe you will have the opportunity
to add those features in a future story. For now, we will keep it simple and
quickly deliver a little more value to our users.

__Acceptance criteria__:
- [x] New images that are added show up on the homepage.
- [x] These images are persisted if the browser is closed or even if the
  server is restarted.
- [x] Images are not displayed wider than 400px.
- [x] Newest images appear first.

__Dependencies__:
- Save Image Link

### PR DESCRIPTION
**Pull request by ragurney**
Closes #3

### CODE REVIEW

File: app/views/images/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<h1> Images Homepage </h1>  
+
+<% @image_list.each do |img| %> 
+  <div>
```
Comment (aaronpanch) [inline app/views/images/index.html.erb]: Having a `div` without any classes is sort of a "bad smell" -- you're probably using it to make sure each image appears on it's own line, and that's fine for now.  In subsequent stories this should be refactored.

File: app/views/images/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    <img class="image-tile" src= <%= img.url %>>
```
Comment (aaronpanch) [inline app/views/images/index.html.erb]: There's a special rails helper that makes generating image tags easier: `image_tag`. Take a look at [this](http://api.rubyonrails.org/classes/ActionView/Helpers/AssetTagHelper.html#method-i-image_tag)

File: app/views/images/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<h1> Images Homepage </h1>  
+
+<% @image_list.each do |img| %>
```
Comment (bboe) [inline app/views/images/index.html.erb]: `img` or `image`?

File: db/migrate/20160819175531_add_timestamps_to_images.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class AddTimestampsToImages < ActiveRecord::Migration[5.0]
+  def change
+    add_column :images, :created_at, :datetime
```
Comment (bboe) [inline db/migrate/20160819175531_add_timestamps_to_images.rb]: Should these fields be nullable?

File: test/controllers/images_controller_test.rb [hunk 5]
```diff
@@ -40,4 +40,26 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
     assert_response :unprocessable_entity
     assert_equal 'Url cannot be empty', flash[:error]
   end
+
+  test 'image index should list all images in db' do
```
Comment (bboe) [inline test/controllers/images_controller_test.rb]: What if there are no images?

**Comment by bboe** [comment]
@ragurney additional comments on this PR. Please combine these updates with the other PR.

cc @aaronpanch

File: app/views/images/index.html.erb (hunk 3 above)
Comment (ragurney) [inline app/views/images/index.html.erb]: 0e5a9a1

File: db/migrate/20160819175531_add_timestamps_to_images.rb (hunk 4 above)
Comment (ragurney) [inline db/migrate/20160819175531_add_timestamps_to_images.rb]: dbce7e5

File: test/controllers/images_controller_test.rb (hunk 5 above)
Comment (ragurney) [inline test/controllers/images_controller_test.rb]: 1f67dad

**Pull request merged**